src/F2_autoreport.py

```diff
@@ -248,6 +248,91 @@ def insert_log_messages(log_dict):
     )
 
 
+def insert_plot(
+    x_data,
+    y_data,
+    plot_type,
+    plot_title=None,
+    x_axis_name=None,
+    y_axis_name=None,
+    id_plot=None,
+    print_only=False,
+):
+    """ Function to create plots using the Plotly library and replace the plots already generated by the
+    matplotlib library, in the web app version of the auto-report.
+    :param print_only: bool, default value is False
+    :param plot_title: str
+    :param y_axis_name: str
+    :param x_axis_name: str
+    :param id_plot: str, id of the graph. needed to work on the plot later
+    :param x_data: list, or pandas series
+    :param y_data: list, or pandas series, or list of lists
+    :param plot_type: type of the plot (line/bar/pie)
+    :return: a plot object
+    """
+    fig = go.Figure()
+    if plot_type == "line":
+        fig.add_trace(
+            go.Scatter(
+                x=x_data,
+                y=y_data,
+                mode="lines",
+                line=dict(color="royalblue", width=2.5),
+            )
+        )
+    styling_dict = dict(
+        showgrid=True,
+        gridwidth=1,
+        zeroline=True,
+        mirror=True,
+        title_font=dict(size=18, family="Courier", color="black"),
+    )
+    fig.update_layout(
+        title=plot_title,
+        xaxis_title=x_axis_name,
+        yaxis_title=y_axis_name,
+        template="simple_white",
+        xaxis=styling_dict,
+        yaxis=styling_dict,
+    )
+    rendered_plots = [
+        html.Img(
+            className="plots-print",
+            src="data:image/png;base64,{}".format(
+                base64.b64encode(fig.to_image(format="png", width=1000)).decode(),
+            ),
+        )
+    ]
+    if print_only is False:
+        rendered_plots.append(
+            dcc.Graph(className="no-print", id=id_plot, figure=fig, responsive=True)
+        )
+    return html.Div(children=rendered_plots)
+
+
+def ready_plots(df_pd, dict_of_labels):
+    """
+    """
+    list_of_keys = list(df_pd.columns)
+    list_of_keys.remove("timestamp")
+    plots = []
+    for component in list_of_keys:
+        comp_id = component + "-plot"
+        plots.append(
+            insert_plot(
+                x_data=df_pd["timestamp"],
+                y_data=df_pd[component],
+                plot_type="line",
+                plot_title=dict_of_labels[component],
+                x_axis_name="Time",
+                y_axis_name="kW",
+                id_plot=comp_id,
+                print_only=False,
```

Comment:
`print_only` also need to be an argument of ready_plots, otherwise we can't change its value :)